lib/Dialect/D2M/Transforms/InsertDstRegisterAccess.cpp

```diff
@@ -464,14 +467,37 @@ struct D2MInsertDstRegisterAccessRewriter final
               AffineMap dstAccessMap, ValueRange dstAccessIndices) {
             auto dstLoad = rewriter.create<affine::AffineLoadOp>(
                 loc, dst, dstAccessMap, dstAccessIndices);
+            Value valueToStore = dstLoad.getResult();
+
+            // Insert unrealized conversion cast if destination CB type differs
+            // from dst type
+            auto cbType = mlir::cast<MemRefType>(cb.getType());
```

Comment:
I could be wrong, but I think this could also potentially be a tensor, so `dyn_cast` if so? Or require memref in the op verifier? Same comment about `dstType` below.